reframe/core/pipeline.py

```diff
@@ -1516,7 +1517,7 @@ def run(self):
                 self._copy_to_stagedir(os.path.join(self._prefix,
                                                     self.sourcesdir))
 
-        super().run()
+        super().run.__wrapped__(self)
```

Comment:
Brava! Didn't know about that, but that's exactly what we want! I think it works, because we're using `@functools.wraps` in the `@_run_hooks` decorator.